addons/ui/fnc_openLobbyManager.sqf

```diff
@@ -0,0 +1,400 @@
+        };
+    };
+
+    [] call _fnc_collapse;
+}];
+
+private _ctrlButtonOK = _display displayCtrl IDC_OK;
+_ctrlButtonOK ctrlAddEventHandler ["ButtonClick", {
+    collect3DENHistory {
+        params ["_ctrlButtonOK"];
+
+        private _display = ctrlParent _ctrlButtonOK;
+        private _ctrlSlots = _display displayCtrl IDC_LM_SLOTS;
+
+        private _groups = [];
+        private _selected = [];
+
+        // set3DENSelected takes flattened array
+        {
+            _selected append _x;
+        } forEach get3DENSelected "";
+
+        for "_sideIndex" from 0 to ((_ctrlSlots tvCount []) - 1) do {
+            for "_groupIndex" from 0 to ((_ctrlSlots tvCount [_sideIndex]) - 1) do {
+                private _groupPath = [_sideIndex, _groupIndex];
+                private _group = _ctrlSlots getVariable (_ctrlSlots tvData _groupPath);
+                _groups pushBack _group;
+
+                private _leader = leader _group;
+                private _multipleUnits = count units _group > 1;
+
+                private _units = [];
+
+                for "_unitIndex" from 0 to ((_ctrlSlots tvCount _groupPath) - 1) do {
+                    private _unitPath = [_sideIndex, _groupIndex, _unitIndex];
+                    private _unit = _ctrlSlots getVariable (_ctrlSlots tvData _unitPath);
+                    private _description = _unit getVariable QGVAR(description);
+
+                    set3DENSelected [_unit];
+
+                    if (_multipleUnits) then {
+                        do3DENAction "CutUnit";
+                        do3DENAction "PasteUnitOrig";
+                    };
+
+                    _unit = get3DENSelected "Object" select 0;
+                    _units pushBack _unit;
+
+                    add3DENConnection ["Group", _units, _group];
+                    _group selectLeader _leader;
+
+                    _unit set3DENAttribute ["description", _description];
+                };
+            };
+        };
+
+        // Need frame delaying to avoid duplication bugs
+        [_groups, _selected] spawn {
+            params ["_groups", "_selected"];
+
+            private _total = count _groups;
+            startLoadingScreen [localize LSTRING(AdjustingGroupOrder)];
+
+            {
+                private _group = _x;
+                private _callsign = _group getVariable QGVAR(callsign);
+
+                set3DENSelected [_group];
+
+                private _handle = 0 spawn {do3DENAction "CutUnit"};
+                waitUntil {scriptDone _handle};
+
+                private _handle = 0 spawn {do3DENAction "PasteUnitOrig"};
+                waitUntil {scriptDone _handle};
+
+                _group = get3DENSelected "Group" select 0;
+                _group set3DENAttribute ["groupID", _callsign];
+
+                progressLoadingScreen ((_forEachIndex + 1) / _total);
+            } forEach _groups;
+
+            set3DENSelected _selected;
+            endLoadingScreen;
```

Comment:
This ends all loading screens. might theoretically happen that other mods start other loadingscreens inbetween while your scheduled code runs. Though it's unlikely.